ts/sagas/wallet.ts

```diff
@@ -74,26 +81,79 @@ import {
   feeExtractor,
   getFavoriteWalletId
 } from "../store/reducers/wallet/wallets";
-import { Transaction } from "../types/pagopa";
 import { Wallet } from "../types/pagopa";
+import {
+  SessionResponse,
+  Transaction,
+  TransactionListResponse,
+  WalletListResponse
+} from "../types/pagopa";
 import {
   UNKNOWN_AMOUNT,
   UNKNOWN_PAYMENT_REASON,
   UNKNOWN_RECIPIENT
 } from "../types/unknown";
 
+// allow refreshing token this number of times
+const MAX_TOKEN_REFRESHES = 2;
+
 function* fetchTransactions(
-  loadTransactions: () => Promise<ReadonlyArray<Transaction>>
+  pagoPaClient: PagoPaClient,
+  token: string,
+  retries: number = MAX_TOKEN_REFRESHES
 ): Iterator<Effect> {
-  const transactions: ReadonlyArray<Transaction> = yield call(loadTransactions);
-  yield put(transactionsFetched(transactions));
+  if (retries === 0) {
```

Comment:
the API client already retries [on error with an exponential backoff](https://github.com/teamdigitale/italia-app/blob/master/ts/utils/fetch.ts#L19)

Reply:
But this retry logic is specific to pagoPA, in that -- before retrying -- it starts a new session (via `fetchPagoPaToken`, which uses `startSession`'s API call) if a 401 is received